scripts/generateAssetData/utils/portals.ts

```diff
@@ -1,31 +1,171 @@
 import type { ChainId } from '@shapeshiftoss/caip'
 import { ASSET_NAMESPACE, bscChainId, toAssetId } from '@shapeshiftoss/caip'
-import type { Asset } from '@shapeshiftoss/types'
+import type { Asset, AssetsByIdPartial } from '@shapeshiftoss/types'
+import { isSome } from '@shapeshiftoss/utils'
 import axios from 'axios'
 import qs from 'qs'
 import { getAddress, isAddressEqual, zeroAddress } from 'viem'
 import { CHAIN_ID_TO_PORTALS_NETWORK } from 'lib/market-service/portals/constants'
-import type { GetTokensResponse, TokenInfo } from 'lib/market-service/portals/types'
+import type {
+  GetPlatformsResponse,
+  GetTokensResponse,
+  PlatformsById,
+  TokenInfo,
+} from 'lib/market-service/portals/types'
 
+import generatedAssetData from '../../../src/lib/asset-service/service/generatedAssetData.json'
 import { colorMap } from '../colorMap'
 import { createThrottle } from '.'
 
+const assets = generatedAssetData as unknown as AssetsByIdPartial
+
+const PORTALS_API_KEY = process.env.REACT_APP_PORTALS_API_KEY
+
+export const getPortalTokens = async (nativeAsset: Asset): Promise<Asset[]> => {
+  const PORTALS_API_KEY = process.env.REACT_APP_PORTALS_API_KEY
+  if (!PORTALS_API_KEY) throw new Error('REACT_APP_PORTALS_API_KEY not set')
+
+  const portalsPlatforms = await fetchPortalsPlatforms()
+  const chainId = nativeAsset.chainId
+  const explorerData = {
+    explorer: nativeAsset.explorer,
+    explorerAddressLink: nativeAsset.explorerAddressLink,
+    explorerTxLink: nativeAsset.explorerTxLink,
+  }
+
+  const portalsTokens = await fetchPortalsTokens(chainId)
+  return portalsTokens.map(token => {
+    const assetId = toAssetId({
+      chainId,
+      assetNamespace: chainId === bscChainId ? ASSET_NAMESPACE.bep20 : ASSET_NAMESPACE.erc20,
+      assetReference: token.address,
+    })
+    const asset = assets[assetId]
+
+    const platform = portalsPlatforms[token.platform]
+    const isPool = Boolean(platform && token.tokens?.length) || undefined
+
+    const iconOrIcons = (() => {
```

Comment:
we should not be using the generated assets for populating icon(s). this will not populate images for any assets that don't already exist in the generated assets json:
```
"eip155:1/erc20:0xdf666a5370fb4b21672f9145b29a56d589cc91fd": {
    "explorer": "https://etherscan.io",
    "explorerAddressLink": "https://etherscan.io/address/",
    "explorerTxLink": "https://etherscan.io/tx/",
    "color": "#3C3C3C",
    "name": "Convex vETHETH Pool",
    "precision": 18,
    "symbol": "cvxvETHETH",
    "chainId": "eip155:1",
    "assetId": "eip155:1/erc20:0xdf666a5370fb4b21672f9145b29a56d589cc91fd",
    "relatedAssetKey": null,
    "isPool": true
  },
  ```

Reply:
gm a69391a3cf
https://jam.dev/c/4178ece3-7a75-462e-85fd-4d5d15c05cf7 
<img width="676" alt="image" src="https://github.com/user-attachments/assets/9d049815-6e56-48cb-986b-538ca4f45f3b">